File: src/main/java/io/github/arieslab/pages/base/ImprimirPage.java
package io.github.arieslab.pages.base;

import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.protocol.http.WebApplication;

import java.io.File;

public class ImprimirPage extends WebPage {
    private static final long serialVersionUID = 1L;
    public ImprimirPage() {}

    protected static String resolveRealPath() {
        try {
            var app = (WebApplication) org.apache.wicket.Application.get();
            var realPath = app.getServletContext().getRealPath("");
            if (realPath != null) return realPath;
        } catch (Exception ignored) {}
        return System.getProperty("user.dir");
    }
}

File: src/main/java/io/github/arieslab/pages/base/ImprimirPage.kt
package io.github.arieslab.pages.base

import org.apache.wicket.markup.html.WebPage

open class ImprimirPage : WebPage() {

    companion object {
        protected fun resolveRealPath(): String {
            return try {
                val app = org.apache.wicket.protocol.http.WebApplication.get()
                val realPath = app.servletContext.getRealPath("")
                if (realPath != null) realPath else System.getProperty("user.dir")
            } catch (_: Exception) {
                System.getProperty("user.dir")
            }
        }
    }
}
